src/main/kotlin/net/atos/zac/enkelvoudiginformatieobject/EnkelvoudigInformatieObjectLockService.kt

```diff
@@ -0,0 +1,79 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Lifely
+ * SPDX-License-Identifier: EUPL-1.2+
+ */
+package net.atos.zac.enkelvoudiginformatieobject
+
+import jakarta.enterprise.context.ApplicationScoped
+import jakarta.inject.Inject
+import jakarta.persistence.EntityManager
+import jakarta.persistence.PersistenceContext
+import jakarta.transaction.Transactional
+import jakarta.transaction.Transactional.TxType.REQUIRED
+import jakarta.transaction.Transactional.TxType.SUPPORTS
+import net.atos.client.zgw.drc.DrcClientService
+import net.atos.client.zgw.zrc.ZRCClientService
+import net.atos.client.zgw.zrc.model.Zaak
+import net.atos.zac.enkelvoudiginformatieobject.model.EnkelvoudigInformatieObjectLock
+import net.atos.zac.util.UriUtil
+import nl.lifely.zac.util.AllOpen
+import nl.lifely.zac.util.NoArgConstructor
+import java.util.Optional
+import java.util.UUID
+
+@ApplicationScoped
+@Transactional(SUPPORTS)
+@AllOpen
+@NoArgConstructor
+class EnkelvoudigInformatieObjectLockService @Inject constructor(
+    private val drcClientService: DrcClientService,
+    private val zrcClientService: ZRCClientService
+) {
+    @PersistenceContext(unitName = "ZaakafhandelcomponentPU")
+    private lateinit var entityManager: EntityManager
+
+    @Transactional(REQUIRED)
+    fun createLock(informationObjectUUID: UUID, idUser: String): EnkelvoudigInformatieObjectLock =
+        EnkelvoudigInformatieObjectLock().apply {
+            enkelvoudiginformatieobjectUUID = informationObjectUUID
+            userId = idUser
+            lock = drcClientService.lockEnkelvoudigInformatieobject(informationObjectUUID)
+            entityManager.persist(this)
+        }
+
+    fun findLock(informationObjectUUID: UUID): Optional<EnkelvoudigInformatieObjectLock> {
```

Comment:
I think especially in Kotlin code (but even in a lot of the ZAC Java code) we should get rid of the use of `Optional`. It is misused now in many places I think. In Kotlin it should really not be needed anymore, unless we still need to support integration with Java code maybe. In this case the function should probably just return null if it cannot find the lock. Or maybe a specific exception, if this particular case is exceptional but I guess that's not the case here.